OvsR_classifier_train_test.py

```diff
@@ -37,135 +40,256 @@ def roc_calculation(fpr, tpr, roc_auc, y_test, y_score, n_species):
+for ind, color in zip(range(len(keys)), colors):
     plt.plot(fpr_test[ind], tpr_test[ind], color=color,
-             label='ROC curve of' + keys[ind] +' (area = {0:0.2f})'
+             label='ROC curve of' + keys[ind] + ' (area = {0:0.2f})'
              ''.format(roc_auc_test[ind]))
 
+plt.legend()
+
 f2.tight_layout()
 
-plt.savefig('ROC_AUC.pdf')
+plt.savefig('ROC_AUC_OvR.pdf')
+
+# new figure
+f3 = plt.figure(figsize=[15, 15], constrained_layout=True)
+# plot confusion matrix test
+ax3 = plt.subplot(1, 1, 1)
+im3 = ax3.imshow(corr_df, plt.get_cmap('coolwarm'))
+ax3.figure.colorbar(im3, ax=ax3,)
+ax3.set(xticks=np.arange(corr_df.shape[1]),
+        yticks=np.arange(corr_df.shape[0]),
+        xticklabels=train_test_columns, yticklabels=train_test_columns,
+        title='Correlation matrix',
+        )
+# Rotate the tick labels and set their alignment.
+plt.setp(ax3.get_xticklabels(), rotation=45, ha="right",
+    rotation_mode="anchor")
+
+#add values in the middle of the cell
+for i in range(corr_df.values.shape[0]):
+        for j in range(corr_df.values.shape[1]):
+            ax3.text(j, i, format(corr_df.values[i, j], '.2f'),
+                    ha="center", va="center",
+                    color="black",fontsize=4)
+
+plt.xticks(fontsize = 5)
+plt.yticks(fontsize = 5)
+
+plt.savefig('correlation_matrix_OvR.pdf')
+
+# plot new confusion matrix for multilabel
+new_conf_matr_train = multilabel_confusion_matrix(y_train_df, y_pred_train)
+new_conf_matr_test = multilabel_confusion_matrix(y_test_df, y_pred_test)
+# normalize confusion matrix
+new_conf_matr_train = new_conf_matr_train.astype(
+    'float32') / new_conf_matr_train.sum(axis=1)[:, np.newaxis]
+new_conf_matr_test = new_conf_matr_test.astype(
+    'float32') / new_conf_matr_test.sum(axis=1)[:, np.newaxis]
+conf_matr_list = [new_conf_matr_train, new_conf_matr_test]
+
+# new figure and plot confusion matrixes
+f4, axes = plt.subplots(nrows=2, ncols=5, figsize=[35, 35], constrained_layout=True)
+for i in range(2):
+    for j in range(5):
+        ax = axes[i, j]
+        ticks = [keys[j], 'background']
+        cm = conf_matr_list[i][j]
+        im = ax.imshow(cm, plt.get_cmap('Blues'))
+        ax.figure.colorbar(im, ax=ax)
+        ax.set(xticks = np.arange(cm.shape[1]), yticks = np.arange(cm.shape[0]), xticklabels = ticks, yticklabels = ticks)
+        for ind_i in range(cm.shape[0]):
+            for ind_j in range(cm.shape[1]):
+                ax.text(ind_j, ind_i, format(cm[ind_i, ind_j], '.2f'),
+                    ha="center", va="center",
+                    color="black", fontsize=4)
+
+plt.xticks(fontsize = 5)
+plt.yticks(fontsize = 5)
+
+f4.tight_layout()
+
+plt.savefig('confusion_matrix_of_species_OvrR.pdf')
+
+#adding distribution prob.
+for key,prob in zip(keys,range(len(keys))):
+    train_df['prob_{0}'.format(key)] = y_proba_train[:, prob]
+    test_df['prob_{0}'.format(key)] = y_proba_test[:, prob]
+
+#plot distribution of probabilities
+for key in keys:
```

Comment:
Substitute with:
```py
#plot distribution of probabilities
for key in keys:
    fighist = plt.figure()
    plt.hist(train_df.loc[train_df[key] == 1]['prob_{0}'.format(key)], color = 'r',
    alpha =0.5, bins = 50, histtype='stepfilled', density=True,
    label = '{0}_train'.format(key), log=True)
    # plt.hist(test_df.loc[test_df[key] == 1]['prob_{0}'.format(key)], color = 'b',
    # alpha =0.5, bins = 50, histtype='stepfilled', density=True,
    # label = '{0}_test'.format(key))
    hist, bins = np.histogram(test_df.loc[test_df[key] == 1]['prob_{0}'.format(key)].values, bins = 50, density = True )
    print(bins)
    scale = len(test_df) / sum(hist)
    err = np.sqrt(hist * scale) / scale
    center = (bins[:-1] + bins[1:]) / 2
    plt.errorbar(center, hist, yerr=err, fmt='o', c='r')
    fighist.savefig('probability_distribution_of_{0}_OvR.pdf'.format(key))
```

Reminder: for each probability we should put the distributions for all the species (to be added)
Use same color for all the probability of the same species (i.e. pions always with the same color), and a different color for the different species (i.e. pions red, kaon blue, proton green etc)